graphql_compiler/macros/__init__.py

```diff
@@ -176,7 +181,16 @@ def get_schema_for_macro_definition(schema):
     Returns:
         GraphQLSchema usable for writing macros
     """
-    raise NotImplementedError()  # TODO(bojanserafimov): Implement
+
+    required_macro_directives = (MacroEdgeDefinitionDirective, MacroEdgeTargetDirective)
+
+    # pylint: disable=protected-access
+    schema_for_macro_definition = copy(schema)
+    schema_for_macro_definition._directives = list(chain(
```

Comment:
What if the macros are already defined in the schema?

Reply:
In the next commit, the list is first converted to a set, to eliminate duplicates.